### DISCUSSION TITLE
Why are test runs failing without any error after adding this logger?

### DISCUSSION THREAD

**Discussion by AArnott** [Tyrrrz/GitHubActionsTestLogger#37] [Q&A] [answered]
I'm adding your test logger to my repo template for spreading around a bunch more repos:
https://github.com/AArnott/Library.Template/pull/319

But it's causing all non-Windows build agents to fail without any error message [like this](https://github.com/AArnott/Library.Template/actions/runs/12324148948/job/34401133228?pr=319).

Any idea why?

**Comment by Tyrrrz** [comment]
No clue, there seems to not be any error message produced in the process. Does removing `IncludeAssets="runtime"` do anything?

**Reply by AArnott** [reply]
No, that attribute doesn't matter. I tried both ways. In fact, this problem repos in both repos that I have tried it in.

**Reply by AArnott** [reply]
It appears to be the combination of **3** switches.

This fails on linux:

    dotnet test --no-build --collect "Code Coverage;Format=cobertura" --logger GitHubActions --diag "obj/_artifacts/test_logs/diag.log;TraceLevel=info"

With the error: 
>  Library.Tests test net8.0 failed with 1 error(s) (3.9s)
    /usr/share/dotnet/sdk/9.0.100/Microsoft.TestPlatform.targets(48,5): error MSB6006: "dotnet" exited with code 139.

But each of these pass (each missing one of the 3 switches used above in the failing case):

    dotnet test --no-build --collect "Code Coverage;Format=cobertura" --logger GitHubActions        

    dotnet test --no-build --collect "Code Coverage;Format=cobertura" --diag "obj/_artifacts/test_logs/diag.log;TraceLevel=info"

    dotnet test --no-build --logger GitHubActions --diag "obj/_artifacts/test_logs/diag.log;TraceLevel=info"

**Reply by Tyrrrz** [reply]
I have not used the `--diag` option so don't really know what to advise there, sorry

**Reply by AArnott** [reply]
I worked around the exit code 1 failure by [suppressing the `--diag` switch when running in GitHub Actions on linux/mac](https://github.com/AArnott/Library.Template/pull/319/files#diff-4f0855940da424a8a7d87b00d65952360ea3d24fc357a2ae51506885ff9f7f96). It always adds `--logger GitHubActions` when under GitHub Actions, however.

But when testing the annotations from the logger, I see it only works on Windows agents. I have arranged for unique test failures on each agent, but only the Windows agents' failures result in annotations, as can be seen [here](https://github.com/AArnott/Library.Template/actions/runs/12345598828?pr=319).

Is that a known issue?

**Reply by Tyrrrz** [reply]
No, it is not. Sounds like something is wrong with your setup.

Here's a working example for reference: https://github.com/Tyrrrz/CliWrap/actions/runs/12285513560

**Reply by AArnott** [reply, marked as answer]
It turns out that your logger doesn't record annotations on mac/linux when `--collect "Code Coverage;Format=cobertura"` or `--settings some.runsettings` is specified on the CLI as well. So in the end, I got it to work by adding a few more condition checks to avoid using those switches on GitHub Actions (while retaining them everywhere else).

So yay, I'm unblocked and can use your logger. But it would sure be nice to find out why it (or the test runner?) is so fragile.

**Reply by AArnott** [reply]
Well shoot. I'm going to have to roll this back because once my template repo merged into one of my real repos (with a lot of tests), the test step fails with a crash in the runner, with this message:

> Data collector 'Blame' message: Data collector caught an exception of type 'System.ArgumentException': 'Process with an Id of 1980 is not running.'. More details: Blame: Creating hang dump failed with error...

I haven't been able to get a callstack for this exception. I really like the annotations that this logger offers, but I think I'll have to revisit and try again in a few months or longer, because this has introduced too much instability and complexity to my test scripts to try to make work.

**Reply by Tyrrrz** [reply]
I am using it with `--settings some.runsettings` myself. I also collect coverage, although not specifically with `cobertura`.

https://github.com/Tyrrrz/.github/blob/b37ad17e1525b7a27ab8aa9baa7c16a2e38d363c/.github/workflows/nuget.yml#L100-L108

If I did have to guess, I'd assume it's the `--diag` switch that's causing this.

**Reply by AArnott** [reply]
Thanks for reviewing.

I already took the `--diag` switch out (as was required to get past the "exit 1" with no error message problem. But even without that switch, and with changes to my code coverage and test settings switches to match yours, I still got the `ArgumentException` that crashes the test runner (and only on Windows IIRC) and fails the overall step in the workflow, once I applied it to a real repo with a decent set of tests.
